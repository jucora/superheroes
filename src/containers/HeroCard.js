import React from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { addCurrentHero, addCurrentOption } from "../actions/index";

export const HeroCard = (props) => {
  const { id, name, image, addCurrentHero, hero, addCurrentOption } = props;
  return (
    <div className="heroCard">
      <div className="cardImageContainer">
        <h1>{name}</h1>
        <img src={image} width="300" height="300" alt={`${name}`} />
        <hr />

        <Link
          to={{ pathname: `/details/${id}` }}
          style={{ textDecoration: "none" }}
        >
          <button
            onClick={() => {
              addCurrentHero(hero);
              addCurrentOption("appearance");
            }}
            type="button"
          >
            Appearance
          </button>
        </Link>
        <Link
          to={{ pathname: `/details/${id}` }}
          style={{ textDecoration: "none" }}
        >
          <button
            onClick={() => {
              addCurrentHero(hero);
              addCurrentOption("biography");
            }}
            type="button"
          >
            Biography
          </button>
        </Link>
        <Link
          to={{ pathname: `/details/${id}` }}
          style={{ textDecoration: "none" }}
        >
          <button
            onClick={() => {
              addCurrentHero(hero);
              addCurrentOption("powerstats");
            }}
            type="button"
          >
            Powerstats
          </button>
        </Link>
      </div>
    </div>
  );
};

const matchDispatchToProps = (dispatch) => ({
  addCurrentHero: (hero) => {
    dispatch(addCurrentHero(hero));
  },
  addCurrentOption: (option) => {
    dispatch(addCurrentOption(option));
  },
});

HeroCard.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  addCurrentHero: PropTypes.func.isRequired,
  hero: PropTypes.objectOf(PropTypes.any).isRequired,
  addCurrentOption: PropTypes.func.isRequired,
};

export default connect(null, matchDispatchToProps)(HeroCard);
