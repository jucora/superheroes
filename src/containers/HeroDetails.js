import React from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";

const HeroDetails = (props) => {
  const { option } = props;
  const { match } = props;
  const { id } = match.params;
  const { heroResults } = props;
  const currentHero = heroResults[id];

  const createOptionResults = () => {
    window.scrollTo(0, 250);

    if (option === "appearance") {
      const { appearance } = currentHero;
      const { gender } = appearance;
      return (
        <div>
          <p>Gender: {gender}</p>
          <p>Eye Color: {appearance["eye-color"]}</p>
          <p>Hair Color: {appearance["hair-color"]}</p>
          <p>Height: {`${appearance.height[0]} - ${appearance.height[1]}`}</p>
          <p>Race: {appearance.race}</p>
          <p>weight: {`${appearance.weight[0]} - ${appearance.weight[1]}`}</p>
        </div>
      );
    }
    if (option === "biography") {
      const { biography } = currentHero;
      return (
        <div>
          <p>
            Aliases:{" "}
            {biography.aliases.forEach((alias) => ({
              alias,
            }))}
          </p>
          <p>Alignment: {biography.alignment}</p>
          <p>Alter-egos: {biography["alter-egos"]}</p>
          <p>First-appearance: {biography["first-appearance"]}</p>
          <p>Full-name: {biography["full-name"]}</p>
          <p>Place-of-birth: {biography["place-of-birth"]}</p>
          <p>Publisher: {biography.publisher}</p>
        </div>
      );
    }
    if (option === "powerstats") {
      const { powerstats } = currentHero;
      return (
        <div>
          <p>Combat: {powerstats.combat}</p>
          <p>Durability: {powerstats.durability}</p>
          <p>Intelligence: {powerstats.intelligence}</p>
          <p>Power: {powerstats.power}</p>
          <p>Speed: {powerstats.speed}</p>
          <p>Strength: {powerstats.strength}</p>
        </div>
      );
    }
    return true;
  };

  return (
    <div className="optionInfo">
      <div className="left">
        <p>{currentHero.name}</p>

        <img src={currentHero.image.url} alt={currentHero.name} />
      </div>

      <div className="right">{createOptionResults()}</div>
    </div>
  );
};

HeroDetails.propTypes = {
  heroResults: PropTypes.arrayOf(PropTypes.object).isRequired,
  location: PropTypes.objectOf(PropTypes.any).isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string.isRequired,
    }),
  }).isRequired,
};

const mapStateToProps = (state) => ({
  heroResults: state.heroReducer.heroResults,
  option: state.heroReducer.option,
  currentHero: state.heroReducer.currentHero,
});

export default connect(mapStateToProps, null)(HeroDetails);
