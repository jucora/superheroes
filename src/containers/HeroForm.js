import React from "react";
import { connect } from "react-redux";
import { updateHeroes } from "../actions/index";
import heroApi from "../api/heroApi";

class HeroForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = { heroName: "" };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  handleSubmit() {
    const { updateHeroes } = this.props;
    const { heroName } = this.state;
    if (heroName === "") return;
    heroApi.getByName(heroName).then((data) => {
      if (data) {
        updateHeroes(data);
      }
    });
  }

  render() {
    const { heroName } = this.state;
    return (
      <div>
        <div className="searchHeroInput">
          <h3 className="searchTitle">SEARCH BY NAME</h3>
          <input
            value={heroName}
            id={heroName}
            name="heroName"
            type="text"
            placeholder="Hero Name"
            className=""
            onChange={this.handleChange}
          />

          <button
            className="submitButton"
            type="submit"
            value="SUBMIT"
            onClick={this.handleSubmit}
          >
            SEARCH
          </button>
        </div>
      </div>
    );
  }
}

const matchDispatchToProps = (dispatch) => ({
  updateHeroes: (results) => {
    dispatch(updateHeroes(results));
  },
});

export default connect(null, matchDispatchToProps)(HeroForm);
