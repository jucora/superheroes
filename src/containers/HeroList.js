import React from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import HeroCard from "./HeroCard";

const displayInput = () => {
  const input = document.querySelector(".searchHeroInput");
  if (input) {
    input.style.display = "flex";
  }
};

export const HeroList = (props) => {
  displayInput();
  const { heroResults } = props;
  return (
    <div className="heroList">
      {heroResults.map((hero, index) => (
        <HeroCard
          id={index}
          key={hero.id}
          name={hero.name}
          image={hero.image.url}
          hero={hero}
        />
      ))}
    </div>
  );
};

HeroList.propTypes = {
  heroResults: PropTypes.arrayOf(PropTypes.object).isRequired,
};

const mapStateToProps = (state) => ({
  heroResults: state.heroReducer.heroResults,
});

export default connect(mapStateToProps, null)(HeroList);
