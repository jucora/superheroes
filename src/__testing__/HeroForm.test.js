import React from "react";
import { render, cleanup } from "@testing-library/react";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { HeroForm } from "../containers/HeroForm";

afterEach(cleanup);

configure({ adapter: new Adapter() });

describe("HeroForm main structure", () => {
  it("should take a snapshot", () => {
    const originalError = console.error;
    console.error = jest.fn();

    const { asFragment } = render(<HeroForm />);
    expect(asFragment(<HeroForm />)).toMatchSnapshot();
    console.error = originalError;
  });
});
