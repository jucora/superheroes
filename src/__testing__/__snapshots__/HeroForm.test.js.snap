// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroForm main structure should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="searchHeroInput"
    >
      <h3
        class="searchTitle"
      >
        SEARCH BY NAME
      </h3>
      <input
        id=""
        name="heroName"
        placeholder="Hero Name"
        type="text"
        value=""
      />
      <button
        class="submitButton"
        type="submit"
        value="SUBMIT"
      >
        SEARCH
      </button>
    </div>
  </div>
</DocumentFragment>
`;
